### ISSUE TITLE
mpv and pulse: EAC3 passthu sets sample rate to 192000 and doesn't work with pulse ao

### ISSUE THREAD

**Issue by space88man** [mpv-player/mpv#9603]
### Important Information
During EAC3 passthru mpv requests 192000 from pulse: this does not work with pulse 15 as it is expecting 48000 (and then will multiply by 4 by itself). Apparently, vlc, will request 48000. 

See: https://gitlab.freedesktop.org/pipewire/pipewire/-/issues/1902. In the provided fix in the pulse emulation layer there is defensive coding to skip the 4x step if the value provided exceeds 48000.

This issue was also reproduced on native pulseaudio and filed here: https://gitlab.freedesktop.org/pulseaudio/pulseaudio/-/issues/1307

Provide following Information:
- mpv version: 0.34.0 (pulseaudio 15.0, pipwire/pipewire-pulseaudio 0.3.40)
- Linux Distribution and Version: Fedora 35
- Source of the mpv binary: distro
- If known which version of mpv introduced the problem:unknown
- Window Manager and version: Gnome 41
- GPU driver and version: NA
- Possible screenshot or video of visual glitches: NA

If you're not using git master or the latest release, update.
Releases are listed here: https://github.com/mpv-player/mpv/releases

### Reproduction steps
Attempt EAC3 passthru; mpv requests 192000, (vlc requests 48000):

```
mpv --audio-device=pulse/alsa_output.pci-0000_01_00.1.hdmi-stereo-extra5 --audio-spdif=eac3 EAC3FILE.mkv, 
[ao] Trying audio driver 'pulse'                                             
[ao] Using preferred device 'alsa_output.pci-0000_07_00.1.hdmi-stereo-extra1' 
[ao/pulse] requested format: 192000 Hz, stereo channels, spdif-eac3
[ao/pulse] Library version: 15.0.0 
[ao/pulse] Proto: 35                                                         
[ao/pulse] Server proto: 4294967295
[ao/pulse] Channel layouts:                                                  
[ao/pulse]  - #fl                                                            
[ao/pulse]  - #fr                                                            
[ao/pulse]  - #fc                                                            
[ao/pulse]  - #lfe                                                           
[ao/pulse]  - #bl                                                            
[ao/pulse]  - #br                  
[ao/pulse]  - #flc                                                           
[ao/pulse]  - #frc                                                           
[ao/pulse]  - #bc                                                            
[ao/pulse]  - #sl                                                            
[ao/pulse]  - #sr                                                            
[ao/pulse]  - #tc         
[ao/pulse]  - #tfl                                                           
[ao/pulse]  - #tfc                                                           
[ao/pulse]  - #tfr                                                           
[ao/pulse]  - #tbl                                                           
[ao/pulse]  - #tbc                                                           
[ao/pulse]  - #tbr                                                           
[ao/pulse] result: stereo          
[ao/pulse] Stream failed. 
```

### Expected behavior

EAC3 passthru works

### Actual behavior

EAC3 passthru fails

### Log file

[mpv-pulse.log](https://github.com/mpv-player/mpv/files/7727614/mpv-pulse.log)

### Sample files
NA - any EAC3 file will work

**Comment by ValZapod** [comment]
This is correct, EAC3 ALWAYS USES 192kHz sample rate and only 2 channels. The underlying sample rate may be 48k and have 7.1 channels with Atmos...

**Comment by ford-prefect** [comment]
The PulseAudio passthrough API is designed so that the client presents the media sample rate, and it opens the device with the appropriate rate/channel count (`rate*4` for EAC3, `channels=8` for TrueHD). Is there a reason MPV must use the multiplied rate here?

**Comment by ValZapod** [comment]
https://github.com/mpv-player/mpv/blob/master/audio/decode/ad_spdif.c#L231 can be fixed for this, but there is a problem of how to populate it.

**Comment by ford-prefect** [comment]
> Always 192k 2 channels for eac3 and mlp package for truehd, no?

It's been a while since I looked at this, so I'm a bit fuzzy on the base container + object audio metadata variants.

AIUI, for "standard" E-AC3, it's 192k for 48 kHz data, 176.4k for 44.1 kHz data, etc. I guess what you're saying is E-AC3 with object audio metadata should always be 192k?

**Comment by ford-prefect** [comment]
> https://github.com/mpv-player/mpv/blob/master/audio/decode/ad_spdif.c#L231 can be fixed for this, but there is a problem of how to populate it.

I guess it is a bit ugly to have audio-device specific code there, so not sure what the right way to deal with this is. PipeWire has a quirk for this case to fix the format mpv provides, which is pushing the ugliness to the audio server. If there is really no way to solve this in mpv, I can add that fix in PulseAudio too.

**Comment by ValZapod** [comment]
> I can add that fix in PulseAudio too.

Yes, please do.

**Comment by ValZapod** [comment]
@ford-prefect Any updates?

**Comment by ford-prefect** [comment]
I am travelling for a bit, so don't have access to my AVR. It might be a few weeks before I can get to this (probably for 17.0).
